repl/scala-2.11/src/main/scala/org/apache/spark/repl/SparkIMain.scala

```diff
@@ -1129,7 +1129,7 @@ class SparkIMain(@BeanProperty val factory: ScriptEngineFactory, initialSettings
 
     def apply(line: String): Result = debugging(s"""parse("$line")""")  {
       var isIncomplete = false
-      currentRun.reporting.withIncompleteHandler((_, _) => isIncomplete = true) {
+      currentRun.parsing.withIncompleteHandler((_, _) => isIncomplete = true) {
```

Comment:
It is harmless, if not beneficial. But this can be a stepping stone towards enabling add jars, because then we have two options 1) back port scala's version of add jars 2) port spark scala 2.10 repl's version of add jars on the fly.

Without this patch such a option does not exist. I agree best thing is to patch scala/scala so that we donot have to do it. So I am working on it, with whatever time I have.